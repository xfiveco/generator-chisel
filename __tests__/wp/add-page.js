const fs = require('fs');
const { answers } = require('./helpers');

describe('WP add-page', () => {
  test('Add pages', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create'],
      answers({ port: 8081 }),
      { interceptWpConfig: true, mockRandomBytes: true },
    );

    const consoleMock = jest.spyOn(console, 'log');

    await global.chiselTestHelpers.runChiselScript([
      'add-page',
      'Page 1',
      'Fancy ąęśćłó/Nested/Name',
    ]);

    global.chiselTestHelpers.normalizeConsoleMockCalls(consoleMock);
    expect(consoleMock.mock.calls).toMatchSnapshot();

    const templates = './wp/wp-content/themes/chisel-test-wp-chisel/templates';
    await global.chiselTestHelpers.expectFilesToMatchSnapshot([
      `${templates}/page-*.twig`,
    ]);

    expect(fs.readFileSync(`${templates}/page-page-1.twig`, 'utf8')).toContain(
      `{% set pageName = 'Page 1' %}`,
    );
    expect(
      fs.readFileSync(
        `${templates}/page-fancy-aesclo-nested-name.twig`,
        'utf8',
      ),
    ).toContain(`{% set pageName = 'Fancy ąęśćłó/Nested/Name' %}`);
  });
});
