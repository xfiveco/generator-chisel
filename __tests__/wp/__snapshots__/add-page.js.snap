// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WP add-page Add pages 1`] = `
Array [
  Array [],
  Array [
    "[92m[1mPage Page 1 created![22m[39m",
  ],
  Array [
    "[1mTemplate:[22m --PROJECT-PATH--/wp/wp-content/themes/chisel-test-wp-chisel/templates/page-page-1.twig",
  ],
  Array [
    "[1mURL:[22m http://localhost:8081/?page_id=4",
  ],
  Array [],
  Array [
    "[92m[1mPage Fancy Ä…Ä™Å›Ä‡Å‚Ã³/Nested/Name created![22m[39m",
  ],
  Array [
    "[1mTemplate:[22m --PROJECT-PATH--/wp/wp-content/themes/chisel-test-wp-chisel/templates/page-fancy-aesclo-nested-name.twig",
  ],
  Array [
    "[1mURL:[22m http://localhost:8081/?page_id=5",
  ],
  Array [],
]
`;

exports[`WP add-page Add pages 2`] = `
Array [
  "./wp/wp-content/themes/chisel-test-wp-chisel/templates/page-fancy-aesclo-nested-name.twig",
  "./wp/wp-content/themes/chisel-test-wp-chisel/templates/page-page-1.twig",
  "./wp/wp-content/themes/chisel-test-wp-chisel/templates/page-plugin.twig",
]
`;
